### ISSUE TITLE
Android > 5 -- unable to find files on device

### ISSUE THREAD

**Issue by eb1** [adapt-it/adapt-it-mobile#264]
Adapt It Mobile Version: 1.0 beta 8
Operating System / version: Android / 6.0.1
Device: Samsung Galaxy 5
Severity: 1 / unable to proceed
Description: From Bruce -

Phone also has an SD card, as well as its internal storage. She read that the document has to be available for importing from the Document or Downloads folder, on card or on phone's internal storage. Your documentation also says that .SFM and .txt files are acceptable. She put a trial document, the gospel of Mark in .SFM format (which is utf-8 PlainText I think) in each of those possible locations. She could not get your current version to load in to the app.  It says "No avaliable files found on device. Please copy files onto the device and try again."

She copied over the AI-BasicConfiguration.aic file, but that made no difference either. Changing the Mark file's .SFM extension to .txt make no difference either.

Files were copied via File Commander to:
Internal storage : Documents and Downloads folder
External storage : SD card : Documents and Downloads folder

**Comment by eb1** [comment]
screen shots from phone:

![ai mob dic not found](https://user-images.githubusercontent.com/1458944/31803036-2a5c307a-b517-11e7-8907-ca3844925467.png)
![ai mob main scr](https://user-images.githubusercontent.com/1458944/31803037-2a6d04ae-b517-11e7-836b-41daba3fd39b.png)

**Comment by adaptitbruce** [comment]
Ester tried an Anroid tablet - she got the same problem. --Bruce

**Comment by eb1** [comment]
For the external directories, we _might_ be running into this open issue in the Cordova / Android libs: https://issues.apache.org/jira/browse/CB-10883. Note that the existing code is working on iOS, as well as on a similar emulated system (Galaxy 5) running on Android 5.x.

I'm still looking at other possibilities to address this. Techno-babble: if worse comes to worse, I might be able to create a plugin to expose the location of the [Android Environment object](https://developer.android.com/reference/android/os/Environment.html) from native code up to Cordova. I've got a question out to Stack Overflow asking if there's already a plugin out there that does this; if not, I'll attempt this approach.

**Comment by eb1** [comment]
Update on this issue:

It appears that we were running into a runtime permissions issue, which is new in Android 6 and above. Rather than returning a permissions error, `cordova-file-plugin` was silently failing to return a result set for us to parse. Cordova's android platform is currently in a state of flux, partly to better support runtime permissions. In the meantime, I was able to fix the issue by adding a diagnostic plugin and requesting the permission at AIM startup -- if it has not already been granted.

I've built an updated AIM app and sent out a link to test (the fix has only been verified in an emulator). Once we get results back, we'll officially ship another beta milestone release.
